### ISSUE TITLE
[v7-alpha.21] - pageIndex is reset if set to nonzero

### ISSUE THREAD

**Issue by yyuurreecc5** [tannerlinsley/react-table#1450]
**Describe the bug**
If you set the pageIndex value to other than zero, then the useEffect hook works 2 times for the first time with the value set, and the second with the pageIndex = 0.
**To Reproduce**
1. Go to https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/pagination-controlled
2. On line 51, replace the pageIndex value 0 with 15 (or any other...). 
const tableState = useTableState({ pageIndex: 15 }) 
3. Click the "refresh page" button on the right side of the codesandbox screen.
4. Further, we see that the data fetched twice, the first time with a value of 15, and the second time with a value of 0. Value of the indexPage changed to 0 instead of 15. In the same time "Go to page" input has value = 16.

**Expected behavior**
I expect that the effect hook and dataFetch work once with the value initially set and value of the indexPage is equal to the value set in.

**Codesandbox!**
https://codesandbox.io/s/tannerlinsleyreact-table-pagination-controlled-qbxgh

**Comment by tannerlinsley** [comment]
I'll investigate this asap. Thanks for your help!

**Comment by tannerlinsley** [comment]
First thing to note here is that a `pageIndex` of `15` is in fact page `16` since `pageIndex` is zero-index based. So this is expected and correct.

**Comment by tannerlinsley** [comment]
Looks like the current page being displayed is incorrect thought :(

**Comment by tannerlinsley** [comment]
Yeah. I think I found the culprit. It's the layout effect here:

```js
  useLayoutEffect(() => {
    setState(
      old => ({
        ...old,
        pageIndex: 0,
      }),
      actions.pageChange
    )
  }, [setState, rowDep, filters, groupBy, sortBy])
```

Clearly this shouldn't run on mount...

**Comment by yyuurreecc5** [comment]
> First thing to note here is that a `pageIndex` of `15` is in fact page `16` since `pageIndex` is zero-index based. So this is expected and correct.

Yes, I just meant that this is the right behavior.

**Comment by tannerlinsley** [comment]
Gotcha :)

**Comment by tannerlinsley** [comment]
Should be fixed in d50ec58

**Comment by yyuurreecc5** [comment]
> Should be fixed in [d50ec58](https://github.com/tannerlinsley/react-table/commit/d50ec588bceb74f7bbf4ab547609a8e312931207)

Thanks a lot, now everything works correctly :+1:
